controllers/metal3machine_controller.go

```diff
@@ -151,6 +145,19 @@ func (r *Metal3MachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, errors.Wrapf(err, "failed to create helper for managing the machineMgr")
 	}
 
+	// Return early if the BMMachine or Cluster is paused.
+	if util.IsPaused(cluster, capm3Machine) {
```

Comment:
this means that the BMH will be paused also if the pause annotation is set on the m3machine. Is that desired behaviour ?

Reply:
I suppose it is? If the CAPM3Machine is paused, shouldn't the associated BMH be? What is your opinion?

Reply:
I don't think that m3machine and bmh should be tied on that. you might want to pause the m3machine reconciliation, but keep BMO reconciling the BMH to get the possible state changes for example

Reply:
Changed the code to remove the tie.